kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -534,43 +429,28 @@ protected Optional<LogSegment> getLogSegmentForVersion(
       Engine engine,
```

Comment:
Correct me if I'm wrong, can't this now return `LogSegment` instead of `Optional<LogSegment>` since we take care of all the missing/DNE table cases in the inner functions? (and never return Optional.empty())

I think this change would help make this code a lot clearer as well as it's hard to track what Optional.empty() means throughout the code... (personal gripe I've had in the past)

Reply:
> Correct me if I'm wrong, can't this now return LogSegment instead of Optional<LogSegment>

I believe it can return LogSegment now, right. I was hoping to keep this refactor tightly scoped to consolidation of the listing files. But I can include that change, sure.